eval.go

```diff
@@ -442,8 +443,12 @@ func (d *indexData) scoreFile(fileMatch *FileMatch, doc uint32, mt matchTree, kn
 	// the matches.
 	fileMatch.addScore("fragment", maxFileScore, opts.DebugScore)
 
-	// Prefer docs with several top-scored matches.
-	fileMatch.addScore("repetition-boost", scoreRepetitionFactor*float64(repetitions), opts.DebugScore)
+	if repetitions != 0 {
+		// Prefer docs with several top-scored matches. We use log_2 (bits.Len) to
+		// prevent the repetitions overriding other factors. In this way it acts
+		// more like a tie break.
+		fileMatch.addScore("repetition-boost", scoreRepetitionFactor*float64(bits.Len(repetitions)), opts.DebugScore)
```

Comment:
Curious why you pick `bits.Len` instead of `math.Log2`? Zoekt already uses `math.Log` for scoring `traction`. Doesn't bits.Len lead to a step-like behavior, boosting 7 repetitions just as much as 4?

math.Log (or Log2) would give an almost linear or at least strictly monotonous behavior for small repetitions and still dampen the boost for large number of repetitions.

Reply:
Because I over worried about perf. You are right, I should probably just call log.

Reply:
@stefanhengl What do you reckon? This will be a lot of calls to this. The other use of log happens at index time not per filematch.

```
BenchmarkBoostLen-32            1000000000               0.2085 ns/op
BenchmarkBoostLog-32            191207702                6.110 ns/op
```

Reply:
I would be more worried about the behaviour of the step-like function than about the performance. With 6ns/op you have to call this a lot to make a dent on a search? I think we can merge this, but we have to watch out for cases of good files being ranked lower because we haven't crossed the next 2x threshold to get a higher boost.